packages/blade/src/components/Modal/Modal.web.tsx

```diff
@@ -0,0 +1,238 @@
+  return css`
+    box-shadow: ${theme.elevation.highRaised};
+    position: fixed;
+    transform: translate(-50%, -50%);
+    opacity: ${isVisible ? 1 : 0};
+    animation: ${isVisible ? entry : exit}
+      ${castWebType(makeMotionTime(theme.motion.duration.xmoderate))}
+      ${
+        isVisible
+          ? castWebType(theme.motion.easing.entrance.revealing)
+          : castWebType(theme.motion.easing.exit.revealing)
+      }};
+  `;
+});
+
+const Modal = ({
+  isOpen = false,
+  children,
+  onDismiss,
+  initialFocusRef,
+  size = 'small',
+  accessibilityLabel,
+}: ModalProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const [footerHeight, setFooterHeight] = useState(0);
+  const { isMounted, isVisible } = usePresence(isOpen, {
+    transitionDuration: theme.motion.duration.xmoderate,
+    initialEnter: true,
+  });
+
+  // required by floating ui to handle focus
+  const { refs, context } = useFloating({
+    open: isMounted,
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const defaultInitialFocusRef = React.useRef<any>(null);
+  const originalFocusElement = React.useRef<HTMLElement | null>(null);
+
+  const returnFocus = React.useCallback(() => {
+    if (!originalFocusElement.current) return;
+    originalFocusElement.current.focus();
+    // After returning focus we will clear the original focus
+    // Because if modal can be opened up via multiple triggers
+    // We want to ensure the focus returns back to the most recent triggerer
+    originalFocusElement.current = null;
+  }, [originalFocusElement]);
+
+  const focusOnInitialRef = React.useCallback(() => {
+    if (!initialFocusRef) {
+      // focus on close button
+      defaultInitialFocusRef.current?.focus();
+    } else {
+      // focus on the initialRef passed by the user
+      initialFocusRef.current?.focus();
+    }
+  }, [initialFocusRef]);
+
+  React.useEffect(() => {
+    if (isOpen) {
+      // set the original focus element where the focus will return to after closing the modal
+      originalFocusElement.current =
+        originalFocusElement.current ?? (document.activeElement as HTMLElement);
+      // focus on an element on Modal, if initialFocusRef is not passed, focus on the close button
+      focusOnInitialRef();
+    }
+  }, [isOpen, focusOnInitialRef]);
+
+  React.useEffect(() => {
+    // Return focus to the element that originally had it
+    if (!isOpen) {
+      returnFocus();
+    }
+  }, [isOpen, returnFocus]);
+
+  const modalContext = React.useMemo(
+    () => ({
+      isOpen,
+      close: onDismiss,
+      defaultInitialFocusRef,
+      isVisible,
+      footerHeight,
+      setFooterHeight,
+    }),
+    [isOpen, onDismiss, defaultInitialFocusRef, isVisible, footerHeight, setFooterHeight],
+  );
+  const handleKeyDown = (event: React.KeyboardEvent): void => {
+    // close modal on escape key press
+    if (event?.key === 'Escape' || event?.code === 'Escape') {
```

Comment:
We need to check both on `key` and `code`?

Reply:
Better to check both. During OTPInput implementation I had read that some older browser versions send either of these.